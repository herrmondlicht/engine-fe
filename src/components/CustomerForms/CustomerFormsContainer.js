import React, { useState, useEffect, useCallback } from "react";

import engineAPI from "../../utils/apiRoutes/apiRoutes";

import WholeFormView from "./WholeForm/WholeFormView";

export const createWholeFormContainer = ({ engineAPI }) =>
  function WholeFormContainer() {
    const [modelsList, changeModelsList] = useState([]);
    const [hasFilledForm, setHasFilledForm] = useState(false);
    const [carForm, setCarForm] = useState({
      id: null,
      model: "",
      make: "",
      year: "",
    });

    const [customerForm, setCustomerForm] = useState({
      id: null,
      documentNumber: "",
      fullName: "",
      address: "",
      email: "",
      phone: "",
    });

    const [customerCarForm, setCustomerCarForm] = useState({
      id: null,
      licensePlate: "",
      carId: null,
      customerId: null,
      displacement: "",
      color: "",
    });

    const [error, setError] = useState("");

    async function fetchModelsFromMaker() {
      const modelsList = await engineAPI.cars.get();
      changeModelsList(modelsList.data.data);
    }

    async function insertNewCar() {
      if (!carForm.id) {
        const response = await engineAPI.cars.post({
          data: {
            model: carForm.model,
            make: carForm.make,
            manufacture_year: carForm.year,
          },
        });
        setCarForm((prevState) => ({
          ...prevState,
          id: response.data.data.id,
        }));
        return response.data.data;
      }
      return carForm;
    }

    async function insertNewCustomer() {
      if (!customerForm.id) {
        const response = await engineAPI.customers.post({
          data: {
            document_number: customerForm.documentNumber,
            fullname: customerForm.fullName,
            address: customerForm.address,
            email: customerForm.email,
            phone: customerForm.phone,
          },
        });
        setCustomerForm((prevState) => ({
          ...prevState,
          id: response.data.data.id,
        }));
        return response.data.data;
      }
      return customerForm;
    }

    async function insertNewCustomerCar(carId, customerId) {
      if (!customerCarForm.id) {
        const response = await engineAPI.customers.post({
          urlExtension: `${customerId}/cars`,
          data: {
            license_plate: customerCarForm.licensePlate,
            car_id: carId,
            displacement: customerCarForm.displacement,
            color: customerCarForm.color,
          },
        });
        setCustomerCarForm((prevState) => ({
          ...prevState,
          id: response.data.data.id,
        }));
        return response.data.data;
      }
      return customerCarForm;
    }

    async function sendForm() {
      try {
        const { id: carId } = await insertNewCar();
        const { id: customerId } = await insertNewCustomer();
        await insertNewCustomerCar(carId, customerId);
        setError("");
        setHasFilledForm(true);
      } catch (e) {
        console.error(e);
        setError("Não foi possível adicionar os dados");
      }
    }

    const getChangeFormKeyToForm = (setForm) => (key) => (event) => {
      if (event.persist) event.persist();
      setForm((prev) => ({ ...prev, [key]: event?.target?.value }));
    };

    useEffect(() => {
      fetchModelsFromMaker();
    }, []);

    const changeCarFormForKey = useCallback(
      getChangeFormKeyToForm(setCarForm),
      [setCarForm]
    );
    const changeCustomerFormForKey = useCallback(
      getChangeFormKeyToForm(setCustomerForm),
      [setCustomerForm]
    );
    const changeCustomerCarFormForKey = useCallback(
      getChangeFormKeyToForm(setCustomerCarForm),
      [setCustomerCarForm]
    );

    return (
      <>
        <WholeFormView
          changeCarFormForKey={changeCarFormForKey}
          changeCustomerFormForKey={changeCustomerFormForKey}
          changeCustomerCarFormForKey={changeCustomerCarFormForKey}
          modelsList={modelsList}
          customerForm={customerForm}
          customerCarForm={customerCarForm}
          carForm={carForm}
          error={error}
          sendForm={sendForm}
        />
      </>
    );
  };

export default createWholeFormContainer({ engineAPI });
