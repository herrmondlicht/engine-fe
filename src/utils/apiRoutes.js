import * as axios from "axios";

export const requestMaker = (
  { method, url, isLocked },
  { storage = window.localStorage } = {}
) => ({ data, params }) => {
  const headers = {};
  if (isLocked) headers.Authorization = storage.getItem("token");
  return axios({
    headers,
    method,
    url: `${process.env.REACT_APP_API_SERVICE}/api${url}`,
    data,
    params
  });
};

const routes = {
  login: {
    url: "/session/login",
    isLocked: false
  }
};

const routesWithRequests = Object.keys(routes).reduce(
  (routesReduced, routeKey) => ({
    ...routesReduced,
    [routeKey]: {
      get: requestMaker({
        method: "get",
        isLocked: routes[routeKey].isLocked,
        url: routes[routeKey].url
      }),
      post: requestMaker({
        method: "post",
        isLocked: routes[routeKey].isLocked,
        url: routes[routeKey].url
      })
    }
  }),
  {}
);

export default routesWithRequests;
