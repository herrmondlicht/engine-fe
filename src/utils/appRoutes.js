import AddNewCar from "../components/CarAndCustomerForm/AddCarAndCustomer";

const routes = {
  addnewcar: {
    name: "Add new car",
    path: "addnewcar",
    Component: AddNewCar,
    isLocked: true,
    exact: true
  },
  home: {
    name: "Add new car",
    path: "/",
    Component: AddNewCar,
    isLocked: true,
    exact: true
  }
};

export default routes;
