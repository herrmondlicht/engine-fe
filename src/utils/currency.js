const NOT_DIGIT = /\D/g;

export const toBRL = value => {
  if (isNaN(value)) {
    return "-";
  }
  return Number(value.toString()).toLocaleString("pt-br", {
    style: "currency",
    currency: "BRL",
  });
};

export const fromBRL = value => {
  if (value === undefined || value === null) {
    return "-";
  }
  return fixDecimalPoint(value.toString());
};

export const handleCurrencyFieldChange = value => {
  const fixedValue = fixDecimalPoint(value.toString()).split("");
  if (fixedValue[0] === ".") {
    fixedValue.splice(0, 0, "0");
  }
  if (fixedValue[0] === "0" && fixedValue.length > 4) {
    fixedValue.shift();
  }
  return toBRL(fixedValue.join(""));
};

export const sanitizeNumber = value => value?.replace(NOT_DIGIT, "");

const fixDecimalPoint = value => {
  if (!value) {
    return "0.00";
  }
  const sanitizedValueRaw = sanitizeNumber(value.toString());
  const valueRawArray = sanitizedValueRaw.split("");
  valueRawArray.splice(-2, 0, ".");
  const valueWithDecimalPoint = valueRawArray.join("");

  return valueWithDecimalPoint;
};
