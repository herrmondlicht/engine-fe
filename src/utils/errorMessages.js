export const getErrorMessage = (errorType) => {
  switch (errorType) {
  case "":
    return "";
  case "CAR_FORM":
    return "Não foi possível adicionar o carro. Verifique os dados e tente novamente";
  case "CUSTOMER_FORM":
    return "Não foi possível adicionar o cliente. Verifique os dados e tente novamente";
  case "CUSTOMER_CAR_FORM":
    return "Não foi possível adicionar o carro do cliente. Verifique os dados e tente novamente";
  default:
    return "Não foi possível completar a ação";
  }
};
