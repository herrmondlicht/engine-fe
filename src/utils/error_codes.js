export const ERROR_CODES = {
  DUP00001: "DUP00001",
  AUTH00001: "AUTH00001",
  INTERNAL_ERROR: "INTERNAL_ERROR",
};

export const MAPPED_ERROR = {
  ER_DUP_ENTRY: {
    status: 400,
    code: ERROR_CODES.DUP00001,
    error: "Duplicated object in database.",
  },
  UnauthorizedError: {
    status: 401,
    code: ERROR_CODES.AUTH00001,
    error: "Invalid token",
  },
  invalid_token: {
    status: 401,
    code: ERROR_CODES.AUTH00001,
    error: "Invalid token",
  },
  INTERNAL_ERROR: {
    status: 500,
    code: ERROR_CODES.INTERNAL_ERROR,
    error: "Something went wrong",
  },
};
