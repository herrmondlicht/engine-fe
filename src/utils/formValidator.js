import * as yup from "yup";

yup.setLocale({
  mixed: {
    default: "Não é válido",
    required: "Campo obrigatório",
  },
});

export { yup };
