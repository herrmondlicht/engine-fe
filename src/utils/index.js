export * from "./errorMessages";
export * from "./storage/storageAPI";
export * from "./engineAPI/engineAPI";
export * from "./appRoutes";