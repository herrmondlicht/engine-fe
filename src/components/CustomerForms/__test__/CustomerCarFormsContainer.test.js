import { render, fireEvent, wait } from "@testing-library/react";
import { createCustomerFormsContainer } from "../CustomerFormsContainer";
import { getErrorMessage } from "../../../utils/errorMessages";
import React from "react";
import { stub, assert as sinonAssert } from "sinon";

let engineAPI = {
  cars: {
    post: stub().resolves({ data: { data: {} } }),
    get: stub().resolves({ data: { data: [] } }),
  },
  customers: {
    post: stub().resolves({ data: { data: {} } }),
  },
};

const createTestComponent = (props) => {
  const Component = createCustomerFormsContainer({ engineAPI });
  return render(<Component {...props} />);
};

describe("CustomerCarForms", () => {
  afterEach(() => {
    engineAPI = {
      cars: {
        post: stub().resolves({ data: { data: {} } }),
        get: stub().resolves({ data: { data: [] } }),
      },
      customers: {
        post: stub().resolves({ data: { data: {} } }),
      },
    };
  });

  it("return of render", async () => {
    const { asFragment, getByTestId } = createTestComponent();
    await wait(() => getByTestId("CustomerFormsView_button"));
    expect(asFragment()).toMatchSnapshot();
  });

  it("return of render when the CarForm throws an error", async () => {
    const { getByTestId, queryByText } = createTestComponent();

    await wait(() => getByTestId("CustomerFormsView_button"));
    engineAPI.cars.post.rejects({});
    fireEvent.click(getByTestId("CustomerFormsView_button"));
    await wait(() => queryByText(getErrorMessage("CAR_FORM")));
  });

  it("return of render when the CustomerForm throws an error", async () => {
    engineAPI.customers.post.rejects({});
    const { getByTestId, queryByText } = createTestComponent();
    await wait(() => getByTestId("CustomerFormsView_button"));
    fireEvent.click(getByTestId("CustomerFormsView_button"));
    await wait(() => queryByText(getErrorMessage("CUSTOMER_FORM")));
  });

  it("return of render when the CustomerCarForm throws an error", async () => {
    engineAPI.customers.post.rejects({});
    const { getByTestId, queryByText } = createTestComponent();
    await wait(() => getByTestId("CustomerFormsView_button"));
    fireEvent.click(getByTestId("CustomerFormsView_button"));
    await wait(() => queryByText(getErrorMessage("CUSTOMER_CAR_FORM")));
  });

  it("call API to save the entire form", async () => {
    const expectedArgsCarStub = {
      data: { make: "", model: "", manufacture_year: "", fuel: "" },
    };
    const expectedArgsCustomerForm = {
      data: {
        document_number: "",
        fullname: "",
        address: "",
        email: "",
        phone: "",
      },
    };

    const expectedArgsCustomerCarForm = {
      urlExtension: "1/cars/",
      data: {
        license_plate: "",
        car_id: 1,
        displacement: "",
        color: "",
        customer_id: 1,
      },
    };

    engineAPI.cars.post.resolves({
      data: { data: { id: 1 } },
    });

    engineAPI.customers.post = engineAPI.customers.post.resolves({
      data: { data: { id: 1 } },
    });

    const { getByTestId } = createTestComponent();

    await wait(() => getByTestId("CustomerFormsView_button"));
    fireEvent.click(getByTestId("CustomerFormsView_button"));

    await wait(() => {
      sinonAssert.calledWith(engineAPI.cars.post, expectedArgsCarStub);
      sinonAssert.calledWith(
        engineAPI.customers.post.firstCall,
        expectedArgsCustomerForm
      );
      sinonAssert.calledWith(
        engineAPI.customers.post.secondCall,
        expectedArgsCustomerCarForm
      );
    });
  });
});
