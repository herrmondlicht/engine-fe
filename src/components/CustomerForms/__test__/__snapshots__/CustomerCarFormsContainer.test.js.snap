// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerCarForms return of render 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-sm-12"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Dados do Cliente
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-sm-9"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            CPF (Número)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              maxlength="11"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  CPF (Número)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Telefone (Número)
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Telefone (Número)
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Nome
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Nome
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            E-mail
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="email"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  E-mail
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Endereço
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Endereço
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12"
  >
    <hr
      class="MuiDivider-root makeStyles-divider-6"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-center MuiGrid-grid-sm-12"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1"
      >
        Dados do Veículo
      </p>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-grid-sm-9"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
              for="make"
              id="make-label"
            >
              Marca
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="make"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-202"
                >
                  <span>
                    Marca
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          aria-expanded="false"
          class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon"
          role="combobox"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
              data-shrink="false"
              for="model"
              id="model-label"
            >
              Modelo
            </label>
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
            >
              <input
                aria-autocomplete="list"
                aria-invalid="false"
                autocapitalize="none"
                autocomplete="off"
                class="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
                id="model"
                spellcheck="false"
                type="text"
                value=""
              />
              <div
                class="MuiAutocomplete-endAdornment"
              >
                <button
                  aria-label="Open"
                  class="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
                  tabindex="-1"
                  title="Open"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M7 10l5 5 5-5z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="PrivateNotchedOutline-legendLabelled-202"
                >
                  <span>
                    Modelo
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
      >
        <div
          class="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root makeStyles-selectLabel-233 MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            id="fuel-select"
          >
            Combustível
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="fuel-select fuel"
              class="MuiSelect-root makeStyles-root-232 MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
              id="fuel"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-201"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
            for="year"
            id="year-label"
          >
            Ano
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              id="year"
              type="number"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Ano
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Placa
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              maxlength="8"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Placa
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Cor
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Cor
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4"
      >
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
            data-shrink="false"
          >
            Cilindradas
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
              maxlength="3"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-200 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-202"
              >
                <span>
                  Cilindradas
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root makeStyles-formWrapper-1 MuiGrid-container MuiGrid-item MuiGrid-direction-xs-column MuiGrid-align-items-xs-flex-end MuiGrid-grid-xs-12"
  >
    <span
      class="makeStyles-formErrorMessage-8"
      data-testid="FormErrorMessage_span"
    />
    <div
      class="MuiBox-root MuiBox-root-294"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
        data-testid="CustomerFormsView_button"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-label"
        >
          Confirmar Dados
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
